Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/parser/PRPAIN201306UV02Parser.java

```diff
@@ -99,14 +99,12 @@ public static String getSenderHcid(PRPAIN201306UV02 response) {
                 .getValue().getId().get(0).getRoot();
         }
         //If representedOrganization Id root is null get id from device
-        if (NullChecker.isNullish(id)) {
-            if (response != null && response.getSender() != null
-                && response.getSender().getDevice() != null && response.getSender().getDevice().getId() != null
-                && response.getSender().getDevice().getId().get(0) != null
-                && response.getSender().getDevice().getId().get(0).getRoot() != null
-                && !response.getSender().getDevice().getId().get(0).getRoot().isEmpty()) {
-                id = response.getSender().getDevice().getId().get(0).getRoot();
-            }
+        if (NullChecker.isNullish(id) && response != null && response.getSender() != null
+            && response.getSender().getDevice() != null && response.getSender().getDevice().getId() != null
+            && response.getSender().getDevice().getId().get(0) != null
+            && response.getSender().getDevice().getId().get(0).getRoot() != null
+            && !response.getSender().getDevice().getId().get(0).getRoot().isEmpty()) {
```

Comment:
You can combine the line 105 and 106 to the following: 
CollectionUtils.isNotEmpty(response.getSender().getDevice().getId().get(0).getRoot())

Reply:
CollectionUtils.isNotEmpty method accepts a collection as argument. In this case, response.getReceiver().get(0).getDevice().getId().get(0).getRoot() its a String.